flink-connectors/flink-connector-kinesis/src/test/java/org/apache/flink/streaming/connectors/kinesis/FlinkKinesisConsumerMigrationTest.java

```diff
@@ -0,0 +1,285 @@
+
+		consumerOperator.close();
+		consumerOperator.cancel();
+	}
+
+	@Test
+	public void testRestoreFromFlink11() throws Exception {
+		Properties testConfig = new Properties();
+		testConfig.setProperty(ConsumerConfigConstants.AWS_REGION, "us-east-1");
+		testConfig.setProperty(ConsumerConfigConstants.AWS_CREDENTIALS_PROVIDER, "BASIC");
+		testConfig.setProperty(ConsumerConfigConstants.AWS_ACCESS_KEY_ID, "accessKeyId");
+		testConfig.setProperty(ConsumerConfigConstants.AWS_SECRET_ACCESS_KEY, "secretKey");
+
+		final DummyFlinkKafkaConsumer<String> consumerFunction = new DummyFlinkKafkaConsumer<>(testConfig);
+
+		StreamSource<String, DummyFlinkKafkaConsumer<String>> consumerOperator =
+			new StreamSource<>(consumerFunction);
+
+		final AbstractStreamOperatorTestHarness<String> testHarness =
+			new AbstractStreamOperatorTestHarness<>(consumerOperator, 1, 1, 0);
+
+		testHarness.setTimeCharacteristic(TimeCharacteristic.ProcessingTime);
+
+		testHarness.setup();
+		// restore state from binary snapshot file using legacy method
+		testHarness.initializeStateFromLegacyCheckpoint(
+			getResourceFilename("kinesis-consumer-migration-test-flink1.1-snapshot"));
+		testHarness.open();
+
+		// the expected state in "kafka-consumer-migration-test-flink1.1-snapshot"
+		final HashMap<KinesisStreamShard, SequenceNumber> expectedState = new HashMap<>();
+		expectedState.put(new KinesisStreamShard("fakeStream1",
+				new Shard().withShardId(KinesisShardIdGenerator.generateFromShardOrder(0))),
+			new SequenceNumber("987654321"));
+
+		// assert that state is correctly restored from legacy checkpoint
+		assertNotEquals(null, consumerFunction.getRestoredState());
+		assertEquals(1, consumerFunction.getRestoredState().size());
+		assertEquals(expectedState, consumerFunction.getRestoredState());
+
+		consumerOperator.close();
+		consumerOperator.cancel();
+	}
+
+	// ------------------------------------------------------------------------
+
+	private static String getResourceFilename(String filename) {
+		ClassLoader cl = FlinkKinesisConsumerMigrationTest.class.getClassLoader();
+		URL resource = cl.getResource(filename);
+		if (resource == null) {
+			throw new NullPointerException("Missing snapshot resource.");
+		}
+		return resource.getFile();
+	}
+
+	private static class DummyFlinkKafkaConsumer<T> extends FlinkKinesisConsumer<T> {
+		private static final long serialVersionUID = 1L;
+
+		@SuppressWarnings("unchecked")
+		DummyFlinkKafkaConsumer(Properties properties) {
+			super("test", mock(KinesisDeserializationSchema.class), properties);
+		}
+
+		@Override
+		protected KinesisDataFetcher<T> createFetcher(List<String> streams,
+													  	SourceFunction.SourceContext<T> sourceContext,
+													  	RuntimeContext runtimeContext,
+													  	Properties configProps,
+													  	KinesisDeserializationSchema<T> deserializationSchema) {
+			return mock(KinesisDataFetcher.class);
+		}
+	}
+}
+
+/*
```

Comment:
I think this can be removed, because the legacy checkpoint binary is already maintained by git and should never disappear